core/src/toga/widgets/button.py

```diff
@@ -5,20 +5,19 @@
 import toga
 from toga.handlers import wrapped_handler
 
-from .base import Widget
+from .base import StyleT, Widget
 
 if TYPE_CHECKING:
     from toga.icons import IconContent
 
 
 class OnPressHandler(Protocol):
-    def __call__(self, widget: Button, **kwargs: Any) -> None:
+    def __call__(self, widget: Button, **kwargs: Any) -> object:
         """A handler that will be invoked when a button is pressed.
```

Comment:
So, I have all of the the handler Protocols in existence now...but they still have problems outside of not being compatible with async and generator versions.

### Positional arguments
- If we take `Button` here as an example, users must create a callable where the first positional argument is named `widget`; this is an unnecessary requirement for users
- There are two ways to solve this:
  - Rename the variable to `__widget` and mypy and friends will ignore the argument name
  - Use a positional only signature: `def __call__(self, widget: Button, /, **kwargs: Any) -> object:`
  - This is the go-forward method in Python to denote this....but this brings up the second problem because it requires users to include `**kwargs` in their handler functions

### Enforcement of `kwargs`
- The current signatures are enforcing inclusion of `kwargs` in users' implementations of handlers
- IMHO, I think this is an abuse of the typing system in this example....because it is not that there _are_ unknown arguments, it is that there could _eventually_ be unknown arguments
- In this way, this is prescribing a design pattern that is subjective (albeit reasonable)
- A similar goal could be accomplished by adding a comment to the docstring about forward compatibility via `kwargs`

At this point, I'm a bit past hills I'm willing to die on regarding these Protocols 🥲....but if these arguments are pervasive in some way, this is how I'd update the Protocols for now:
```python
class OnPressHandler(Protocol):
    def __call__(self, widget: Button, /) -> object:
        """A handler that will be invoked when a button is pressed.
        
        Future versions may pass new keyword arguments; to maintain forward compatibility
        with Toga in your handlers, add ``**kwargs`` to your handler definition.
        """
```
Notes:
- As noted above, the final `/` says there's one positional argument of type `Button`
- The return type of `object` allows users to return whatever they want. This is changed from `None` because Toga doesn't actually care and users' functions can do what they want here
- And finally, users' functions can specify arbitrary keyword arguments after the first argument while maintaining typing compliance

Reply:
> So, I have all of the the handler Protocols in existence now...but they still have problems outside of not being compatible with async and generator versions.
> 
> ### Positional arguments
> * If we take `Button` here as an example, users must create a callable where the first positional argument is named `widget`; this is an unnecessary requirement for users
> * There are two ways to solve this:
>   
>   * Rename the variable to `__widget` and mypy and friends will ignore the argument name

Blurgh. I don't like that much.

>   * Use a positional only signature: `def __call__(self, widget: Button, /, **kwargs: Any) -> object:`
>   * This is the go-forward method in Python to denote this....but this brings up the second problem because it requires users to include `**kwargs` in their handler functions
> ### Enforcement of `kwargs`
> * The current signatures are enforcing inclusion of `kwargs` in users' implementations of handlers
> * IMHO, I think this is an abuse of the typing system in this example....because it is not that there _are_ unknown arguments, it is that there could _eventually_ be unknown arguments
> * In this way, this is prescribing a design pattern that is subjective (albeit reasonable)

Agreed... but I'm actually OK with this. We've gently encouraged the use of `**kwargs` in any event handler as a forward compatibility aid; a mechanism that actually enforces this would be a good thing, IMHO.
 
> * A similar goal could be accomplished by adding a comment to the docstring about forward compatibility via `kwargs`

... which will be read by nobody, and adhered to by even less :-)
 
> At this point, I'm a bit past hills I'm willing to die on regarding these Protocols 🥲....

Totally understood :-)

> but if these arguments are pervasive in some way, this is how I'd update the Protocols for now:

FWIW: I'm +1 on the `/` syntax to enforce the named argument; my preference would be to preserve the `**kwargs` so that anyone with noisy type checking is "encouraged" to do the right thing. That gives us a little more wiggle room for future backwards compatibility discussions, because we can add a named keyword argument to the spec for a protocol without breaking backwards compatibility. My experience from Django API definition is that if you have a flexible spec, you don't tie yourself in knots trying to work out how to accomodate a necessary change.